libraries/Microsoft.Bot.Connector/Authentication/Retry.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    public static class Retry
+    {
+        public static async Task<TResult> Run<TResult>(Func<Task<TResult>> task, Func<Exception, int, RetryParams> retryExceptionHandler)
```

Comment:
This method has no unit tests. Seems like it would be easy to test.

Reply:
Forgot to port them, adding them in next push